hack/terraform/aks/main.tf

```diff
@@ -105,7 +90,7 @@ resource "azurerm_kubernetes_cluster" "k8s" {
   identity {
     type = "SystemAssigned"
   }
-
+  kubernetes_version = "1.24.3"
```

Comment:
Remove this.. without this, region default will be picked automatically. Otherwise will be end up updating this version, whenever aks depreciate the support

Reply:
My concern is that if we are not compatible with a latest version, the CI failure might be confusing.

Reply:
CI will break one way or the other if version is removed or we are not compatible. Ideally we want to be compatible with Managed cluster default version, so its fine IMO. We should be able to debug the issue easily.

Reply:
done